Library/Homebrew/cmd/update.rb

```diff
@@ -92,10 +95,25 @@ def update
       end
     end
 
-    if report.empty?
+    # Report changes to Homebrew itself and tapped Git repositories.
+    repo_changes.reject! { |_key, r| r[0] == r[1] }
```

Comment:
Thanks for the feedback! I believe the penalty paid by using `.reject!` is negligible, given that the number of taps will always be a moderate number and given that `pull!` and `report` from `Updater` are much more heavyweight (spawning `git` several times, network access, etc.). Thus, I'd like to keep it for the sake of code clarity and elegance.

The first option (checking before adding an unwanted change) would work, but would make the code uglier, as I would have to replicate the logic in two places and would have to store the result of `.revision_range` in a temporary variable.

I cannot use the second option (skipping inside `.each`) as I rely on `repo_changes.empty?` later on.